src/constraints.jl

```diff
@@ -150,12 +150,64 @@ function constraint_gate_target_condition_compact(qcm::QuantumCircuitModel)
     
     elseif decomposition_type == "approximate"
         JuMP.@constraint(qcm.model, U_var[:,:,max_depth] .== qcm.data["target_gate"][:,:] + qcm.variables[:slack_var][:,:])
-    
+
+    elseif decomposition_type == "exact_optimal_global_phase"
+        if qcm.data["are_gates_real"] 
+            QCO.constraint_gate_target_condition_compact_global_phase_real(qcm)        
+        else
+            QCO.constraint_gate_target_condition_compact_global_phase_complex(qcm)
+        end     
     end
     
     return
 end
 
+function constraint_gate_target_condition_compact_global_phase_real(qcm::QuantumCircuitModel)
+    
+    max_depth   = qcm.data["maximum_depth"]
+    n_r            = size(qcm.data["gates_real"])[1]
+    n_c            = size(qcm.data["gates_real"])[2]
+
+    U_var = qcm.variables[:U_var]
+    
+    ref_nonzero_r, ref_nonzero_c = QCO._get_ref_nonzero_index_of_real_target(qcm.data::Dict{String,Any})
+
+    for i=1:n_r, j=1:n_c
+        if isapprox(qcm.data["target_gate"][i,j], 0, atol=1E-6)
+            JuMP.@constraint(qcm.model, U_var[i,j,max_depth] == 0)
+        else
+            JuMP.@constraint(qcm.model, U_var[i,j,max_depth]*qcm.data["target_gate"][ref_nonzero_r,ref_nonzero_c] == qcm.data["target_gate"][i,j]*U_var[ref_nonzero_r,ref_nonzero_c,max_depth])
+        end
+    end
+    return    
```

Comment:
This function needs to be tested in unit tests. Also, I think this and `constraint_gate_target_condition_compact_global_phase_complex` could be combined into a single function?

Reply:
combined the two global_phase_constraints
unit test now covering both cases